### PR TITLE
Use default view if instrument does not match any pipeline view

### LINKED ISSUE #3224: Default view should not be required when other views passed
### Description

If a user passes a view to the `MeterProvider` during creation and an instrument not matching that view measures values it is not exported. A user is required to pass the default view as well as their custom view to ensure the default behavior on unmatched views.

This is contrary to [the specification](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.13.0/specification/metrics/sdk.md#view):

> * If the `MeterProvider` has one or more `View`(s) registered:
>   * [...]
>   * If the Instrument could not match with any of the registered `View`(s), the
>     SDK SHOULD enable the instrument using the default aggregation and temporality.
>     Users can configure match-all Views using Drop aggregation
>     to disable instruments by default.

### Environment

- Go Version: 1.19
- opentelemetry-go version: 35019d32bd408b7b67dd4ad8511392b66efe7059

### Steps To Reproduce

`main.go`:

```go
package main

import (
	"context"
	"fmt"
	"log"
	"net/http"
	"os"
	"os/signal"

	"github.com/prometheus/client_golang/prometheus"
	"github.com/prometheus/client_golang/prometheus/promhttp"

	otelprom "go.opentelemetry.io/otel/exporters/prometheus"
	"go.opentelemetry.io/otel/sdk/metric"
	"go.opentelemetry.io/otel/sdk/metric/view"
)

func main() {
	ctx := context.Background()

	exporter := otelprom.New()
	v, err := view.New(
		view.MatchInstrumentName("unmatched-name"),
		view.WithRename("bar"),
	)
	if err != nil {
		log.Fatal(err)
	}

	provider := metric.NewMeterProvider(metric.WithReader(exporter, v))
	meter := provider.Meter("example")

	go serveMetrics(exporter.Collector)

	counter, err := meter.SyncFloat64().Counter("foo")
	if err != nil {
		log.Fatal(err)
	}
	counter.Add(ctx, 5)

	ctx, _ = signal.NotifyContext(ctx, os.Interrupt)
	<-ctx.Done()
}

func serveMetrics(collector prometheus.Collector) {
	registry := prometheus.NewRegistry()
	err := registry.Register(collector)
	if err != nil {
		fmt.Printf("error registering collector: %v", err)
		return
	}

	log.Printf("serving metrics at localhost:2222/metrics")
	http.Handle("/metrics", promhttp.HandlerFor(registry, promhttp.HandlerOpts{}))
	err = http.ListenAndServe(":2222", nil)
	if err != nil {
		fmt.Printf("error serving http: %v", err)
		return
	}
}
```

`go.mod`:

```go
go 1.18

require (
	github.com/prometheus/client_golang v1.13.0
	go.opentelemetry.io/otel v1.10.0
	go.opentelemetry.io/otel/exporters/prometheus v0.31.0
	go.opentelemetry.io/otel/metric v0.32.0
	go.opentelemetry.io/otel/sdk v1.10.0
	go.opentelemetry.io/otel/sdk/metric v0.32.0
)

require (
	github.com/beorn7/perks v1.0.1 // indirect
	github.com/cespare/xxhash/v2 v2.1.2 // indirect
	github.com/go-logr/logr v1.2.3 // indirect
	github.com/go-logr/stdr v1.2.2 // indirect
	github.com/golang/protobuf v1.5.2 // indirect
	github.com/matttproud/golang_protobuf_extensions v1.0.1 // indirect
	github.com/prometheus/client_model v0.2.0 // indirect
	github.com/prometheus/common v0.37.0 // indirect
	github.com/prometheus/procfs v0.8.0 // indirect
	go.opentelemetry.io/otel/trace v1.10.0 // indirect
	golang.org/x/sys v0.0.0-20220520151302-bc2c85ada10a // indirect
	google.golang.org/protobuf v1.28.1 // indirect
)
```

```terminal
$ go run .
2022/09/22 09:08:00 serving metrics at localhost:2222/metrics
```

```terminal
# Returns nothing.
$ curl localhost:2222/metrics
```
### Expected behavior

```terminal
$ curl localhost:2222/metrics
# HELP foo
# TYPE foo counter
foo 5
```

### PR DESCRIPTION
**Pull request by MrAlias**
Fix #3224

### CODE REVIEW

**Comment by MrAlias** [comment]
Hmm, looks like I just rebased this into non-existence :facepalm:. Going to try and resurrect

File: sdk/metric/pipeline.go [hunk 1]
```diff
@@ -233,16 +225,40 @@ func (i *inserter[N]) Instrument(inst view.Instrument, instUnit unit.Unit) ([]in
...
-	// TODO(#3224): handle when no views match. Default should be reader
-	// aggregation returned.
+
+	if !matched { // Apply implicit default view if no explicit matched.
+		a, err := i.aggregator(inst)
+		if err != nil {
+			errs.append(err)
+		}
+		if a != nil {
+			aggs = append(aggs, a)
```
Comment (paivagustavo) [inline sdk/metric/pipeline.go:235]: Not sure if this matter much, but should we only add `a` to the `aggs` slice if it was successfully added to the pipeline?

Same applies to the explicit views matched.

**Review by paivagustavo — APPROVED** [review]
👀  just passing around

We should be able to remove this as well: https://github.com/open-telemetry/opentelemetry-go/blob/main/sdk/metric/config.go#L129-L131

File: sdk/metric/pipeline.go (hunk 1 above)
Comment (MrAlias) [reply sdk/metric/pipeline.go:235]: True. I was tracking this as a secondary concern in https://github.com/open-telemetry/opentelemetry-go/issues/3240. I planned to have a follow on PR to https://github.com/open-telemetry/opentelemetry-go/pull/3238 that addressed this.

I'll update here as well and resolve the conflicts in the other.

Comment (MrAlias) [reply sdk/metric/pipeline.go]: Updated.

**Review by MadVikingGod — APPROVED** [review]

**Comment by MrAlias** [comment]
@paivagustavo I reverted https://github.com/open-telemetry/opentelemetry-go/pull/3237/commits/b56efb06a763e7da174f8bfd896d37e244754ccc. It indeed corrected incorrect behavior, but our tests expected this behavior.

The fix for the issue is included in https://github.com/open-telemetry/opentelemetry-go/pull/3251. I will make sure to handle the merge conflict by including the fix in the default view match based on whichever one is merged first.

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**